hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/HoodieFlinkCopyOnWriteTable.java

```diff
@@ -350,18 +352,18 @@ public HoodieRestoreMetadata restore(HoodieEngineContext context, String restore
   protected HoodieMergeHandle getUpdateHandle(String instantTime, String partitionPath, String fileId,
                                               Map<String, HoodieRecord<T>> keyToNewRecords, HoodieBaseFile dataFileToBeMerged) {
     if (requireSortedRecords()) {
-      return new HoodieSortedMergeHandle<>(config, instantTime, this, keyToNewRecords, partitionPath, fileId,
+      return new FlinkCompactorSortedMergeHandle<>(config, instantTime, this, keyToNewRecords, partitionPath, fileId,
```

Comment:
adv of doing this as a `WriteConfig` would be that these new classes need not exist?